### PR TITLE
fix trainer not resetting lightning_optimizers

### LINKED ISSUE #6285: Optimizers are broken with auto_lr_find=True since 1.1.4
## 🐛 Bug

It seems like #5244 (which went out with 1.1.4) caused some bad interaction with `auto_lr_find=True`. 

Specifically, `lightning_optimizers` are now [cached on the Trainer](https://github.com/PyTorchLightning/pytorch-lightning/pull/5244/files#diff-2ac8599fbc86efe739170e1fb2217ac34dbd69da6a84b705a270e6da5ca7a794R262-R267). However, if we update the `lr` with `auto_lr_find=True`, we would expect the optimizers returned from `configure_optimizers` to change -- so that the `lightning_optimizers` need to be updated -- but this is no longer handled because we [no longer re-wrap the optimizers](https://github.com/PyTorchLightning/pytorch-lightning/pull/5244/files#diff-d931a456f290914901d2d1456ab3c85e63b39669d00723d624b988213dccbfdeR110-R115)  in the general case.

The outcome for me is that training just doesnt converge because we're updating the wrong optimizer.

## Please reproduce using the BoringModel

https://colab.research.google.com/drive/1PJGOBSUdl5_-U9O-fvo83V1On6_siwAC?usp=sharing

### To Reproduce

See the colab^ 

### Expected behavior

Training should work!

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.7.1+cu101
	- pytorch-lightning: 1.2.1
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context

1. This was a pretty frustrating bug to track down, it broke training on my model in a super unconnected way and I had to literally `git bisect` both my repo and pytorch-lightning's repo to find it.

2. It's scary to me that the bug seems to have gone unnoticed for so many versions -- does no one use auto_lr_find=True? Are there no test cases checking this combination?

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

When calling `Trainer.fit` multiple times there is a bug where in the second fit the optimization would use the
lightning optimizers from the first model because it is still hanging on to the old list. 

The fix is to reset the list when new optimizers are requested from the model. 
This way the lightning optimizers get re-wrapped when the training loop starts.

Fixes #6285

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -86,7 +86,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed PyTorch Profiler with `emit_nvtx` ([#6260](https://github.com/PyTorchLightning/pytorch-lightning/pull/6260))
 
 
-- Fixed `trainer.test` from `best_path` hangs after calling `trainer.fit`  ([#6272](https://github.com/PyTorchLightning/pytorch-lightning/pull/6272))
+- Fixed `trainer.test` from `best_path` hangs after calling `trainer.fit` ([#6272](https://github.com/PyTorchLightning/pytorch-lightning/pull/6272))
+
+
```
Comment (carmocca) [inline CHANGELOG.md:91]: This is duplicated

```suggestion
```

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]
